packages/ActionBar/src/hooks/useColumnsArrangement/useColumnsArrangement.js

```diff
@@ -31,23 +32,71 @@ function updateHiddenColumnIdFromLocalStorage(localStoragePrefix, columnId, isVi
   }
 }
 
+function hasDifferences(originalArray, newArray) {
+  if (originalArray.length !== newArray.length) return true;
+
+  return difference(originalArray, newArray).length > 0 || difference(newArray, originalArray).length > 0;
+}
+
+function getInitialOrder({ defaultOrder, isLocalStorageEnabled, localStoragePrefix }) {
+  if (!isLocalStorageEnabled) return defaultOrder;
+
+  const cachedOrder = getColumnIdsfromLocalStorage(
+    getLocalStorageKey(localStoragePrefix, savedItem.ORDER),
+    defaultOrder
+  );
+
+  if (hasDifferences(cachedOrder, defaultOrder)) {
+    window.localStorage.setItem(getLocalStorageKey(localStoragePrefix, savedItem.ORDER), JSON.stringify(defaultOrder));
+    return defaultOrder;
+  }
+
+  return cachedOrder;
+}
+
+function getInitialHiddenColumnIds({
+  defaultHiddenColumnIds,
+  defaultOrder,
+  isLocalStorageEnabled,
```

Comment:
what would be the case where isLocalStorageEnable is false and localStoragePrefix has a value?

can you get isLocalStorageEnabled by only checking if the localStoragePrefix exists?

Reply:
Yes, I can, but the value of `isLocalStorageEnabled` defined on line https://github.com/acl-services/paprika/pull/1027/files#diff-077f2f92812dabe83ab01446d6a63f7262db5d31c06e1ddd9333810dace47fd0R88

I just feel pass it in will be easier to read? but maybe not? I can just skip the function call when `isLocalStorageEnabled === false`